icu4j/main/tests/core/src/com/ibm/icu/dev/test/impl/UnitsTest.java

```diff
@@ -36,6 +37,95 @@ public static boolean compareTwoBigDecimal(BigDecimal expected, BigDecimal actua
         return false;
     }
 
+    @Test
+    public void testComplexUnitsConverter() {
+        ConversionRates rates = new ConversionRates();
+        MeasureUnit input = MeasureUnit.FOOT;
+        MeasureUnit output = MeasureUnit.forIdentifier("foot-and-inch");
+        final MeasureUnitImpl inputImpl = MeasureUnitImpl.forIdentifier(input.getIdentifier());
+        final MeasureUnitImpl outputImpl = MeasureUnitImpl.forIdentifier(output.getIdentifier());
+        ComplexUnitsConverter converter = new ComplexUnitsConverter(inputImpl, outputImpl, rates);
+
+        // Significantly less than 2.0.
+        List<Measure> measures = converter.convert(BigDecimal.valueOf(1.9999));
+        assertEquals("measures length", 2, measures.size());
+        assertEquals("1.9999: measures[0] value", BigDecimal.valueOf(1), measures.get(0).getNumber());
+        assertEquals("1.9999: measures[0] unit", MeasureUnit.FOOT.getIdentifier(),
+                measures.get(0).getUnit().getIdentifier());
+
+        assertTrue("1.9999: measures[1] value", compareTwoBigDecimal(BigDecimal.valueOf(11.9988),
+                BigDecimal.valueOf(measures.get(1).getNumber().doubleValue()), BigDecimal.valueOf(0.0001)));
+        assertEquals("1.9999: measures[1] unit", MeasureUnit.INCH.getIdentifier(),
+                measures.get(1).getUnit().getIdentifier());
+
+        // TODO: consider factoring out the set of tests to make this function more
+        // data-driven, *after* dealing appropriately with the memory leaks that can
+        // be demonstrated by this code.
```

Comment:
"... memory leaks that can be demonstrated by the **ICU4C version of** this code."

Reply:
(I haven't checked whether reuse makes any sense in Java, do we ever modify instances - I think not? Given objects are always allocated on the heap, never on the stack, there's less reason to attempt to reuse?)

Reply:
I'm fixing this as part of unicode-org#1337